pathing/config_file_unix.go

```diff
@@ -2,7 +2,42 @@
 
 package pathing
 
+import (
+	"log"
+	"os"
+	"path/filepath"
+)
+
 const (
 	defaultConfigFile = ".packerconfig"
 	defaultConfigDir  = ".packer.d"
 )
+
+func configDir() (path string, err error) {
+	var dir string
+	homedir := os.Getenv("HOME")
+	if homedir == "" {
+		return "", err
+	}
```

Comment:
Since homedir is only being used for the `hasDefaultConfigFileLocation` and the last else cases, do you think we should `return ""` then ? I think this should probably error, or log something, too.

Reply:
Updated with an error and added test case for error.

Reply:
Dope !